DT-6483: Refactor top card

Antiik91 opened the pull request:
## Proposed Changes

  - Top Card is now expandable, clicking the card show additional information about leg's destination.
  - Refactored and renamed components for clarity. 

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

Antiik91 committed:
initial secondary info, navistack logic

Antiik91 committed:
Refactoring

Antiik91 committed:
handle updated messages

Antiik91 committed:
fix typo, edit classname

Antiik91 committed:
Merge branch 'v3' into DT-6483

Antiik91 committed:
NaviTop -> NaviCardContainer

Antiik91 committed:
simpla classname does not need braces

vesameskanen commented on app/component/itinerary/NaviCardContainer.js:
```diff
@@ -98,6 +85,7 @@ function NaviTop(
       if (legChanged) {
         focusToLeg?.(newLeg);
```
if ( newLeg) ...

Antiik91 replied:
I'm confused as why? 

 newLeg is defined like this:   ```const newLeg = realTimeLegs.find(leg => {
      return legTime(leg.start) <= time && time <= legTime(leg.end);
    });``` in useEffect.

 So basically every 10 seconds we get the same leg until we eventually get the new leg. Thats why I introduced the `legChanged` param to handle the situation where the leg is actually changed.

Antiik91 replied:
Done.

vesameskanen commented on app/component/itinerary/NaviCardContainer.js:
```diff
@@ -98,6 +85,7 @@ function NaviTop(
+        setCardExpanded(false);
       }
 
       if (incomingMessages.size || legChanged) {
```
This code block is in wrong level.  Currentleg and newLeg ban be null while waiting, stiull new messages may appear.

Also  if (!focusRef.current && focusToLeg) {}  block is under wrong if block

Antiik91 replied:
Valid point! Will fix this!

vesameskanen commented on app/component/itinerary/NaviCardContainer.js:
```diff
@@ -163,11 +158,21 @@ function NaviTop(
+          />
         );
       } else {
         naviTopContent = (
```
no need to duplicate jsx item, just set legType conditionally

vesameskanen commented on app/component/itinerary/NaviStack.js:
```diff
@@ -4,9 +4,12 @@ import cx from 'classnames';
   return (
-    <div className={cx('info-stack', !show ? 'slide-out' : 'slide-in')}>
+    <div className={cx('info-stack', 'slide-in')} style={{ top }}>
       {messages.map((notification, index) => (
```
style={{'212px'}}  won't work

Antiik91 replied:
True, this will be changed when the PR is out of draft.

vesameskanen commented on app/component/itinerary/NaviCardContainer.js:
```diff
@@ -108,11 +96,18 @@ function NaviTop(
 
+        // handle messages that are updated.
+        const updatedMessages = currActiveMessages.map(msg => {
+          const incoming = incomingMessages.get(msg.id);
```
How about renaming  currActiveMessages  -> previousValidMessages ?

vesameskanen commented:
NaviSecondaryInfo -> naviCardExtension ?

Antiik91 committed:
PR suggestions

Moved messagehandling and focushandling outside of leg handling block

Renamed currActiveMessages to previousValidMessages

legType is determined conditionally, not the whole jsx

Antiik91 committed:
Renamed naviSecondaryInfo -> NaviCardExtension

Antiik91 committed:
change focus only if new leg is defined

Antiik91 committed:
Lollipop colors by mode color

handling of the last leg

Antiik91 committed:
naviStack top position from css

partisaani commented on app/component/itinerary/navigator.scss:
Values defined in `px` could be revisited against [_spacing.scss](https://github.com/HSLdevcom/digitransit-ui/blob/v3/sass/base/_spacing.scss)

Antiik91 replied:
I'll check on this. There is some issues for example card height, that is still hard coded for desings, and there is no corresponding variable in _spacings.scss

partisaani commented on app/component/itinerary/NaviUtils.js:
```diff
@@ -168,3 +169,68 @@ export const getItineraryAlerts = (realTimeLegs, intl, messages) => {
+        break;
+      default:
+        iconOptions.iconId = `icon-icon_${mode.toLowerCase()}-stop-lollipop`;
+    }
```
Complexity could be eased by using an object literal

partisaani commented on app/component/itinerary/NaviCardContainer.js:
```diff
@@ -37,17 +36,9 @@ function NaviTop(
-    return () => clearTimeout(timer);
-  }, []);
-
   useEffect(() => {
```
This looks like a good candidate for a custom hook as the logic inside `useEffect` is all in all quite multifaceted.

Antiik91 replied:
Yeah, this needs to be refactored, but I think it would be more suited for seperate PR

Antiik91 committed:
remove unncesessary object

Antiik91 committed:
refactor: start using _spacing.scss where possible vol 1

vesameskanen reviewed (commented):
Looks good to me. Review comment about object literals is an imporovement, though.

Antiik91 committed:
use object literal
